tempest-testing-internal/src/main/kotlin/app/cash/tempest/testing/internal/TestUtils.kt

```diff
@@ -32,8 +32,12 @@ import java.net.InetSocketAddress
 import java.net.ServerSocket
 import java.net.Socket
 
-fun pickRandomPort(): Int {
-  ServerSocket(0).use { socket -> return socket.localPort }
+fun allocateRandomPort(): ServerSocket {
+  val socket = ServerSocket(0) //use { socket -> return socket.localPort }
+  Runtime.getRuntime().addShutdownHook(
```

Comment:
Just to make sure the port is released in case something fails before it's released in the test server.